components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/util/EndpointUtil.java

```diff
@@ -755,6 +756,11 @@ public static String getUserConsentURL(OAuth2Parameters params, String loggedInU
                     entry.setQueryString(entry.getQueryString() +
                             "&" + PROP_REDIRECT_URI + "=" + URLEncoder.encode(params.getRedirectURI(), UTF_8));
                 }
+
+                if (params != null) {
+                    entry.setQueryString(entry.getQueryString() + "&" + PROP_OIDC_SCOPE +
+                            "=" + URLEncoder.encode(StringUtils.join(getRequestedOIDCScopes(params), "+"), UTF_8));
```

Comment:
Shouldn't need to join with `+` as the URLEncoder should handle it. Lets join with a space as its the normal format. And you can have a unit test to make sure the end url is formatted in the expected manner